### ISSUE TITLE
Estimar métricas via cross_val_predict não é confiável

### ISSUE THREAD

**Issue by ryukinix** [datascience-ufc/asd#13]
Ponto de atenção sobre o cross_val_predict: 
![image](https://user-images.githubusercontent.com/7642878/54074230-c32f3f00-426e-11e9-943c-67fbbc4cae1f.png)

> It is not appropriate to pass these predictions into an evaluation metric. Use cross_validate to measure generalization error.

É necessário como falei anteriormente ter as métricas individuais por modelo. Acredito que você está fazendo uma espécie de Data Leakage aqui passando todo o dataset de treinamento como teste. Como ele irá fazer N estimações ali pelos folds, para alguns modelos a amostra de teste terá sido observada no treinamento.

ref: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html

**Comment by matheus-frota** [comment]
Caraca não prestei atenção nessa parte, vou modificar isso viu @ryukinix, obrigado pelos comentários.
